src/main/java/com/minecolonies/coremod/commands/ActionMenuState.java

```diff
@@ -215,4 +215,61 @@ public int getIntValueForArgument(@NotNull final String argumentName, final int
             return defaultValue;
         }
     }
+
+    @Override
+    public int hashCode()
+    {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + ((actionMenu == null) ? 0 : actionMenu.hashCode());
+        result = prime * result + ((argumentStateByActionArgumentNameMap == null) ? 0 : argumentStateByActionArgumentNameMap.hashCode());
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 8, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)